File: core/crawling/advanced/sitemap_engine.py
from __future__ import annotations
import re

def parse_sitemap(xml_text: str, max_urls: int = 5000):
    urls=sorted(set(re.findall(r"<loc>\s*(.*?)\s*</loc>", xml_text or '', flags=re.IGNORECASE)))
    return {"urls": urls[:max_urls]}

File: src/crawling/advanced/sitemapEngine.ts
/**
 * Converted from Python: core/crawling/advanced/sitemap_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function parseSitemap(xml_text: any, max_urls: any = 5000): any {
  var urls: any = py.sorted(py.toSet(py.reFindall("<loc>\\s*(.*?)\\s*</loc>", py.or2(xml_text, () => ("")), "i")));
  return {"urls": py.slice(urls, null, max_urls)};
}
